### ISSUE TITLE
mathematical formula for categorical cross entropy

### ISSUE THREAD

**Issue by vinayakumarr** [fchollet/keras#6444]
By reading through I came to know mathematical formulation for binary_cross entropy. Could anybody know mathematical formulation for categorical_crossentropy in keras with softmax function?
![untitled](https://cloud.githubusercontent.com/assets/18217467/25556005/1bb51402-2d12-11e7-87ef-a9bad097a858.png)

**Comment by parag2489** [comment]
See [this](http://deeplearning.net/software/theano/library/tensor/nnet/nnet.html#theano.tensor.nnet.nnet.categorical_crossentropy).

**Comment by vinayakumarr** [comment]
what is x, Could you please give a formulae which is in the above format such as binary cross_entopy

**Comment by justinkterry** [comment]
For those who see this, an answer with a human readable formula would actually be great

**Comment by kai0456** [comment]
x is the random event. p(x) is the observed  probability and q(x) is the predicted probability. The formula can be translated to   softmax function directly.

**Comment by EyeBool** [comment]
The equation for categorical cross entropy is
<p align="center"> 
![cross_entropy](https://user-images.githubusercontent.com/26015273/41822460-2ca0a90a-77f0-11e8-9c71-7e88fa6b5c61.gif)
</p>
The double sum is over the observations `i`, whose number is `N`, and the categories `c`, whose number is `C`. The term `1_{y_i \in C_c}` is the indicator function of the `i`th observation belonging to the `c`th category. The `p_{model}[y_i \in C_c]` is the probability predicted by the model for the `i`th observation to belong to the `c`th category. When there are more than two categories, the neural network outputs a vector of `C` probabilities, each giving the probability that the network input should be classified as belonging to the respective category. When the number of categories is just two, the neural network outputs a single probability `\hat{y}_i`, with the other one being `1` minus the output. This is why the binary cross entropy looks a bit different from categorical cross entropy, despite being a special case of it.

**Comment by bolein** [comment]
@EyeBool for the multi-class case (i.e. when there is only one target class) there is no need in the inner sum because it will sum over just one non-zero element. Right?

**Comment by lsegiet** [comment]
@bolein 
This is correct. Assuming, that each y_i belongs to exactly one class, say: C_{y_i}, then we can write: 
<img width="318" alt="Screenshot 2019-03-16 at 17 22 03" src="https://user-images.githubusercontent.com/9467089/54478217-0e69c480-4810-11e9-92f0-c7c9e3e01486.png">

**Comment by mina-aghaei** [comment]
kj

**Comment by lcslima45** [comment]
Can someone help with this question
https://stats.stackexchange.com/questions/575697/how-can-i-get-the-binary-cross-entropy-from-the-cross-entropy-function-for-gans
